ui/src/main/java/edu/wpi/grip/ui/codegeneration/CodeGenerationOptionsController.java

```diff
@@ -0,0 +1,216 @@
+package edu.wpi.grip.ui.codegeneration;
+
+import edu.wpi.grip.core.Pipeline;
+import edu.wpi.grip.core.events.ConnectionAddedEvent;
+import edu.wpi.grip.core.events.ConnectionRemovedEvent;
+import edu.wpi.grip.core.events.ProjectSettingsChangedEvent;
+import edu.wpi.grip.core.settings.ProjectSettings;
+import edu.wpi.grip.core.settings.SettingsProvider;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Inject;
+
+import org.bytedeco.javacpp.opencv_core.Mat;
+
+import java.io.File;
+
+import javafx.application.Platform;
+import javafx.collections.FXCollections;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.StackPane;
+import javafx.stage.DirectoryChooser;
+
+/**
+ * Controller for the code generation options pane.
+ */
+public class CodeGenerationOptionsController {
+
+  @Inject
+  private SettingsProvider settingsProvider;
+  @Inject
+  private EventBus eventBus;
+  @Inject
+  private Pipeline pipeline;
+
+  @FXML
+  private Pane root;
+  @FXML
+  private GridPane optionsGrid;
+  @FXML
+  private ComboBox<Language> languageSelector;
+  @FXML
+  private CheckBox implementVisionPipeline;
+  @FXML
+  private TextField classNameField;
+  @FXML
+  private Label saveLocationField;
+  @FXML
+  private Button browseButton;
+  @FXML
+  private StackPane extrasPane;
+  @FXML
+  private Pane javaControls;
+  @FXML
+  private TextField packageNameField;
+  @FXML
+  private Pane pythonControls;
+  @FXML
+  private TextField moduleNameField;
+
+  private Language language;
+
+  @FXML
+  private void initialize() {
+    languageSelector.setItems(FXCollections.observableArrayList(Language.values()));
+    extrasPane.setVisible(false);
+    root.getProperties().put("controller", this);
```

Comment:
I don't like this, but I can't think of a better way of having the controller accessible from the main window controller